controllers/drplacementcontrol.go

```diff
@@ -462,6 +456,221 @@ func (d *DRPCInstance) getCurrentHomeClusterName(toCluster string, drClusters []
 	return ""
 }
 
+// checkFailoverPrerequisites checks for any failover prerequsites that need to be met on the
+// failoverCluster before initiating a failover.
+// Returns:
+//   - bool: Indicating if prerequisites are met
+//   - error: Any error in determining the prerequisite status
+func (d *DRPCInstance) checkFailoverPrerequisites(curHomeCluster string) (bool, error) {
+	var (
+		met bool
+		err error
+	)
+
+	if isMetroAction(d.drPolicy, d.drClusters, curHomeCluster, d.instance.Spec.FailoverCluster) {
+		met, err = d.checkMetroFailoverPrerequisites(curHomeCluster)
+	} else {
+		met = d.checkRegionalFailoverPrerequisites()
+	}
+
+	if err == nil && met {
+		return true, nil
+	}
+
+	msg := "Waiting for spec.failoverCluster to meet failover prerequsites"
+
+	if err != nil {
+		msg = err.Error()
+
+		rmnutil.ReportIfNotPresent(d.reconciler.eventRecorder, d.instance, corev1.EventTypeWarning,
+			rmnutil.EventReasonSwitchFailed, err.Error())
+	}
+
+	d.setDRPCCondition(&d.instance.Status.Conditions, rmn.ConditionAvailable, d.instance.Generation,
+		d.getConditionStatusForTypeAvailable(), string(d.instance.Status.Phase), msg)
+
+	return met, err
+}
+
+// checkMetroFailoverPrerequisites checks for any MetroDR failover prerequsites that need to be met on the
+// failoverCluster before initiating a failover from the curHomeCluster.
+// Returns:
+//   - bool: Indicating if prerequisites are met
+//   - error: Any error in determining the prerequisite status
+func (d *DRPCInstance) checkMetroFailoverPrerequisites(curHomeCluster string) (bool, error) {
+	met := true
+
+	d.setProgression(rmn.ProgressionWaitForFencing)
+
+	fenced, err := d.checkClusterFenced(curHomeCluster, d.drClusters)
+	if err != nil {
+		return !met, err
+	}
+
+	if !fenced {
+		return !met, fmt.Errorf("current home cluster %s is not fenced", curHomeCluster)
+	}
+
+	return met, nil
+}
+
+// checkRegionalFailoverPrerequisites checks for any RegionalDR failover prerequsites that need to be met on the
+// failoverCluster before initiating a failover.
+// Returns:
+//   - bool: Indicating if prerequisites are met
+//
+// TODO: Write envtests for this function and desendents
+func (d *DRPCInstance) checkRegionalFailoverPrerequisites() bool {
```

Comment:
(future) I think this function and the subsequent functions need to move to the drcluster_mmode.go given that they are common between the DRCluster and DRPlacementControl. To keep it the same as the metro call, I would change it like
```
func (d *DRPCInstance) checkRegionalFailoverPrerequisites() bool {
	return CheckRegionalFailoverPrerequisites(d.vrgs, d.instance.Spec.FailoverCluster, d.log)
}
```

Reply:
Will track and close with the s3 changes, but this is useful and needed.

Reply:
Tracked in https://github.com/RamenDR/ramen/issues/835